bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/action/RoomActions.java

```diff
@@ -61,90 +61,82 @@ public void setThingHandler(@Nullable ThingHandler handler) {
         return (ThingHandler) handler;
     }
 
-    /**
-     * The setThermpoint room thing action
-     */
-    @RuleAction(label = "@text/actionLabel", description = "@text/actionDesc")
-    public void setThermpoint(
-            @ActionInput(name = "setpoint", label = "@text/actionInputSetpointLabel", description = "@text/actionInputSetpointDesc") @Nullable Double temp,
+    @RuleAction(label = "@text/actionSetThermRoomTempSetpointLabel", description = "@text/actionSetThermRoomTempSetpointDesc")
+    public void setThermRoomTempSetpoint(
+            @ActionInput(name = "temp", label = "@text/actionInputSetpointLabel", description = "@text/actionInputSetpointDesc") @Nullable Double temp,
             @ActionInput(name = "endtime", label = "@text/actionInputEndtimeLabel", description = "@text/actionInputEndtimeDesc") @Nullable Long endTime) {
-        setThermpoint(temp, endTime, "MANUAL");
+        CommonInterface roomHandler = handler;
+        if (roomHandler == null) {
+            logger.info("Handler not set for room thing actions.");
+            return;
+        } else if (temp == null) {
+            logger.info("Temperature is required, action ignored");
+            return;
+        } else if (endTime == null) {
+            logger.info("Temperature provided but no endtime given, action ignored");
+            return;
+        }
+
+        try {
+            String roomId = roomHandler.getId();
+            double setpointTemp = temp;
+            long setpointEnd = endTime;
+            SetpointMode setpointMode = SetpointMode.MANUAL;
+            energy.ifPresent(cap -> cap.setRoomThermTemp(roomId, setpointTemp, setpointEnd, setpointMode));
+        } catch (IllegalArgumentException e) {
+            logger.debug("Ignoring setRoomThermpoint command due to illegal argument exception: {}", e.getMessage());
```

Comment:
Would this be caused by a bug or by user input when calling the action? If the latter, perhaps warning would be justified here?

Reply:
IllegalArgumentException will never be thrown, I believe. I removed the try/catch.